pkgs/by-name/sy/syzkaller/package.nix

```
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  go,
+  ncurses,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "syzkaller";
+  version = "0-unstable-2024-01-09";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "syzkaller";
+    rev = "67d7ec0a69193b2acbb8cdb5109d5505a9128ac5";
+    hash = "sha256-hXYyomTPR7T2zJi49xylQdEXALYV6IbNXdHZo/JHqz8=";
+  };
+
+  nativeBuildInputs = [
+    go
+    ncurses
+  ];
+
+  postPatch =
+    let
+      commitDate = lib.concatStringsSep "" (
+        builtins.tail (lib.splitString "-" (lib.removePrefix "0-" finalAttrs.version))
+      );
+      dateString = "${commitDate}-133700"; # commit date + dummy time
+    in
+    ''
+      substituteInPlace Makefile \
+        --replace-fail '$(shell git rev-parse HEAD)' '${finalAttrs.src.rev}' \
+        --replace-fail '$(shell git diff --shortstat)' "" \
+        --replace-fail '$(shell git log -n 1 --format="%cd" --date=format:%Y%m%d-%H%M%S)' '${dateString}' \
+        --replace-fail './bin' "$out/bin"
+    '';
+
+  configurePhase = ''
+    runHook preConfigure
+
+    mkdir -p $out/bin
+
+    export GOCACHE=$TMPDIR/go-cache
+    export GOPATH="$TMPDIR/go"
+
+    runHook postConfigure
+  '';
+
```

I noticed that there is another folder in "$out/bin": 

```
./bin
├── linux_amd64
│   ├── syz-execprog
│   └── syz-executor
├── syz-db
├── syz-manager
├── syz-mutate
├── syz-prog2c
├── syz-repro
└── syz-upgrade

2 directories, 8 files
```

We should probably move those executables up:

```suggestion

  postFixup = ''
    mv $out/bin/linux_amd64/* $out/bin/
    rmdir $out/bin/linux_amd64
  '';
  
```

---

I'd advocate against this, as these are specific to the target platform. One may also configure this portion of the build specifically to build for another platform. (separately to`pkgsCross` and friends)

---

Yeah, you're right.